### PR TITLE
Add autofix to declaration-colon-space-after

### PR DESCRIPTION
**Pull request by superdem**
**Description**
Enable auto fix option for `declaration-colon-space-after`

### CODE REVIEW

**Comment by jeddy3** [comment]
@superdem Thank you for starting on this PR!

I believe you'll need to add a `fixed` property to all of the tests for this rule, [like so](https://github.com/stylelint/stylelint/blob/master/lib/rules/at-rule-name-case/__tests__/index.js).

We can then be confident the code works.

**Comment by superdem** [comment]
Hi @jeddy3 ,
Thanks for your feedback. 
I've updated the test with fixed, you can check again now.

Best!

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 1]
```diff
@@ -164,57 +166,63 @@ testRule(rule, {
...
       description: "no space after mult-line"
     },
     {
-      code: "a { transition:color 1s,\r\n\twidth 2s; }",
+      code: "a { transition: color 1s,\r\n\twidth 2s; }",
       description: "no space after mult-line CRLF"
     },
     {
-      code: "a { transition:\tcolor 1s,\n\twidth 2s; }",
+      code: "a { transition: color 1s,\n\twidth 2s; }",
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Please, revert this change. We need this test.

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 2]
```diff
@@ -164,57 +166,63 @@ testRule(rule, {
       description: "space after mult-line"
     },
     {
-      code: "a { transition:color 1s,\n\twidth 2s; }",
+      code: "a { transition: color 1s,\n\twidth 2s; }",
       description: "no space after mult-line"
     },
     {
-      code: "a { transition:color 1s,\r\n\twidth 2s; }",
+      code: "a { transition: color 1s,\r\n\twidth 2s; }",
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Please, revert this change. We need this test.

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 3]
```diff
@@ -164,57 +166,63 @@ testRule(rule, {
       description: "space after mult-line"
     },
     {
-      code: "a { transition:color 1s,\n\twidth 2s; }",
+      code: "a { transition: color 1s,\n\twidth 2s; }",
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Please, revert this change. We need this test.

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 4]
```diff
@@ -153,7 +154,8 @@ testRule(rule, {
 testRule(rule, {
   ruleName,
   config: ["always-single-line"],
-
+  skipBasicChecks: true,
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Why did you add `skipBasicChecks`?

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 5]
```diff
@@ -87,7 +83,7 @@ testRule(rule, {
 testRule(rule, {
   ruleName,
   config: ["never"],
-
+  fixed: true,
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Should be `fix: true`.

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 6]
```diff
@@ -6,30 +6,20 @@ const { messages, ruleName } = rule;
 testRule(rule, {
   ruleName,
   config: ["always"],
-
+  skipBasicChecks: true,
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Why did you add `skipBasicChecks`?

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 7]
```diff
@@ -6,30 +6,20 @@ const { messages, ruleName } = rule;
...
-      code: "a { color\n: pink }",
-      description: "newline before and space after"
-    },
-    {
-      code: "a { color\r\n: pink }",
-      description: "CRLF before and space after"
-    },
-    {
-      code: "$map:(key:value)",
+      code: "$map: (key:value)",
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Please, revert this change. We need this test.

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 8]
```diff
@@ -6,30 +6,20 @@ const { messages, ruleName } = rule;
...
       description: "SCSS map with no newlines"
     },
     {
-      code: "$list:('value1', 'value2')",
+      code: "$list: ('value1', 'value2')",
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Please, revert this change. We need this test.

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 9]
```diff
@@ -114,34 +110,39 @@ testRule(rule, {
   reject: [
     {
       code: "a { color : pink; }",
+      fixed: "a { color : pink; }",
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Should not be space after a colon.

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 10]
```diff
@@ -114,34 +110,39 @@ testRule(rule, {
...
       code: "a { color:  pink; }",
+      fixed: "a { color:pink; }",
       description: "two spaces after",
       message: messages.rejectedAfter(),
       line: 1,
       column: 11
     },
     {
       code: "a { color :\tpink; }",
+      fixed: "a { color :\tpink; }",
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Should not be space after a colon.

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 11]
```diff
@@ -114,34 +110,39 @@ testRule(rule, {
...
       description: "tab after",
       message: messages.rejectedAfter(),
       line: 1,
       column: 11
     },
     {
       code: "a { color :\npink; }",
+      fixed: "a { color :\npink; }",
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Should not be space after a colon.

File: lib/rules/declaration-colon-space-after/__tests__/index.js [hunk 12]
```diff
@@ -114,34 +110,39 @@ testRule(rule, {
...
       description: "newline after",
       message: messages.rejectedAfter(),
       line: 1,
       column: 11
     },
     {
       code: "a { color :\r\npink; }",
+      fixed: "a { color :\r\npink; }",
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/__tests__/index.js]: Should not be space after a colon.

File: lib/rules/declaration-colon-space-after/index.js [hunk 13]
```diff
@@ -25,6 +25,18 @@ const rule = function(expectation) {
...
     }
 
+    if (context.fix) {
+      const match = /^: /;
+      const replaceWithSpace = ": ";
+      if (expectation === "always" || expectation === "always-single-line") {
+        root.walkDecls(decl => {
+          decl.raws.between.match(match) === null
+            ? (decl.raws.between = replaceWithSpace)
+            : (decl.raws.between = replaceWithSpace);
```
Comment (hudochenkov) [inline lib/rules/declaration-colon-space-after/index.js]: Strange condition. It will give the same results regardless of condition state.

**Comment by ntwb** [comment]
@superdem Just wondering if you have some time to update this PR?
